src/benchmarklib/tpch/tpch_query_generator.cpp

```diff
@@ -52,6 +70,339 @@ std::string TPCHQueryGenerator::get_preparation_queries() const {
+      const auto date = calculate_date(boost::gregorian::date{1995, 03, 01}, 0, date_diff_dist(random_engine));
+
+      parameters.emplace_back("'"s + segment + "'");
+      parameters.emplace_back("'"s + date + "'");
+      parameters.emplace_back("'"s + date + "'");
+      break;
+    }
+
+    case 4 - 1: {
+      static thread_local std::uniform_int_distribution<> date_diff_dist{0, 9};
+      const auto diff = date_diff_dist(random_engine);
+      const auto begin_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff);
+      const auto end_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff + 3);
+
+      parameters.emplace_back("'"s + begin_date + "'");
+      parameters.emplace_back("'"s + end_date + "'");
+      break;
+    }
+
+    case 5 - 1: {
+      const auto region = regions.list[region_dist(random_engine)].text;
+
+      static thread_local std::uniform_int_distribution<> date_diff_dist{0, 4};
+      const auto diff = date_diff_dist(random_engine);
+      const auto begin_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff * 12);
+      const auto end_date = calculate_date(boost::gregorian::date{1993, 01, 01}, (diff + 1) * 12);
+
+      parameters.emplace_back("'"s + region + "'");
+      parameters.emplace_back("'"s + begin_date + "'");
+      parameters.emplace_back("'"s + end_date + "'");
+      break;
+    }
+
+    case 6 - 1: {
+      static thread_local std::uniform_int_distribution<> date_diff_dist{0, 4};
+      const auto diff = date_diff_dist(random_engine);
+      const auto begin_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff * 12);
+      const auto end_date = calculate_date(boost::gregorian::date{1993, 01, 01}, (diff + 1) * 12);
+
+      static std::uniform_real_distribution<> discount_dist{0.02f, 0.09f};
+      const auto discount = discount_dist(random_engine);
+
+      static thread_local std::uniform_int_distribution<> quantity_dist{24, 25};
+      const auto quantity = quantity_dist(random_engine);
+
+      parameters.emplace_back("'"s + begin_date + "'");
+      parameters.emplace_back("'"s + end_date + "'");
+      parameters.emplace_back(std::to_string(discount - 0.01f));
+      parameters.emplace_back(std::to_string(discount + 0.01f));
+      parameters.emplace_back(std::to_string(quantity));
+      break;
+    }
+
+    case 7 - 1: {
+      const auto nation1 = nations.list[nation_dist(random_engine)].text;
+      const auto nation2 = nations.list[nation_dist(random_engine)].text;
+
+      parameters.emplace_back("'"s + nation1 + "'");
+      parameters.emplace_back("'"s + nation2 + "'");
+      parameters.emplace_back("'"s + nation2 + "'");
+      parameters.emplace_back("'"s + nation1 + "'");
+      break;
+    }
+
+    case 8 - 1: {
+      const auto nation_id = nation_dist(random_engine);
+      const auto nation = nations.list[nation_id].text;
+
+      // No idea why the field is called "weight", but it corresponds to the region of a given nation
```

Comment:
"...but it actually is the regionkey of the nation", maybe?

Reply:
Not sure about that. It just so happens to be the same.

Reply:
"corresponds to the region" is vague, "is actually the regionkey of the nation, confusingly named by tpch-dbgen" is precise, at least to me. Up to you.

Reply:
I want it to be vague. ;)
I am not sure if that was actually the intention of tpch-dbgen and if the name should actually be region.